src/HotTable.vue

```diff
@@ -86,6 +98,38 @@
 
         preventInternalEditWatch(this);
       },
+      matchHotMappersSize: function (data: any[][]): void {
+        const rowsToRemove: number[] = [];
+        const columnsToRemove: number[] = [];
+        const indexMapperRowCount = this.hotInstance.rowIndexMapper.getNumberOfIndexes();
+        const indexMapperColumnCount = this.hotInstance.columnIndexMapper.getNumberOfIndexes();
```

Comment:
🤔 Are you sure this will work when `columns` are defined? Or Array of Objects is used or `schema` is defined?

There might be a slight misalignment between getNumberOfIndexes and sourceCount

Reply:
@wojciechczerniak You're right, this should be probably only done for arrays of arrays, modifications on arrays of objects should be done through `updateSettings`. It's impossible to add/remove columns on AoO with the UI though, so it would affect only changes done by reference from the outside.